Framework/src/Triggers.cxx

```diff
@@ -119,9 +120,19 @@ TriggerFcn Never(const Activity& activity)
   };
 }
 
-TriggerFcn EndOfRun(const Activity&)
+TriggerFcn EndOfRun(const std::string& kafkaBrokers, const std::string& topic, const Activity& activity)
 {
-  return NotImplemented("EndOfRun");
+  auto returnedActivity = activity;
+  auto poller = std::make_shared<core::KafkaPoller>(kafkaBrokers, "EOR_postprocessing_group");
+  poller->subscribe(topic);
+  return [poller, returnedActivity]() mutable -> Trigger {
+    for (const auto& record : poller->poll()) {
+      if (proto_parser::isEOR(record.value())) {
+        return { TriggerType::StartOfRun, true, returnedActivity, static_cast<uint64_t>(record.timestamp().msSinceEpoch), "eor" };
```

Comment:
```suggestion
        return { TriggerType::EndOfRun, false, returnedActivity, static_cast<uint64_t>(record.timestamp().msSinceEpoch), "eor" };
```
You used `TriggerType::StartOfRun` by accident here.
Otherwise, the comments below apply to both:
- I would use `last : false`, since it may not be the last time this trigger fires. If `last` is `true`, `tryTrigger` removes the trigger from the list of currently used triggers, which is probably not what we want.
- please fill `Activity` provided to the `Trigger` with the best knowledge of the run that starts or ends.
- please use  the SOSOR and SOEOR times as `timestamp` in the two triggers. To be precise, I mean the same values that ECS calls `"run_start_time_ms"` and `"run_end_time_ms"`. Is `record.timestamp()` exactly that?

Reply:
Having a quick look at the ECS, I assume it's the good timestamp: https://github.com/AliceO2Group/Control/blob/1e83c517a22e5c4d27199f8dac9301e6d18a97cd/core/environment/environment.go#L205

Reply:
hmm, i'm not sure anymore. I think `record.timestamp()` is different than `event.Timestamp`, while the timestamp from the link mentioned earlier goes to the latter.

Reply:
I am sorry I overlooked the event.timestamp. My mistake